### PR TITLE
[RFC] New iteration on clang format - (version 1)

### PR DESCRIPTION
**Pull request by davidlange6**
I revived the clang-format PR from a year ago. Relative to #21479 I changed BinPackParameters: false in order to try to address comments from @slava77

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@makortel @wddgit please also review

File: FWCore/Catalog/interface/InputFileCatalog.h [hunk 1]
```diff
@@ -28,21 +30,24 @@ namespace edm {
...
-    static bool isPhysical(std::string const& name) {
-      return (name.empty() || name.find(':') != std::string::npos);
-    }
-    
+    std::vector<FileCatalogItem> const& fileCatalogItems() const { return fileCatalogItems_; }
+    std::vector<std::string> const& logicalFileNames() const { return logicalFileNames_; }
+    std::vector<std::string> const& fileNames() const { return fileNames_; }
+    std::vector<std::string> const& fallbackFileNames() const { return fallbackFileNames_; }
+    bool empty() const { return fileCatalogItems_.empty(); }
+    static bool isPhysical(std::string const& name) { return (name.empty() || name.find(':') != std::string::npos); }
```
Comment (Dr15Jones) [inline FWCore/Catalog/interface/InputFileCatalog.h]: So it pushed this to one line. I don't find it as readable, but it is OK.

File: FWCore/Common/interface/LuminosityBlockBase.h [hunk 2]
```diff
@@ -37,48 +37,37 @@ namespace edm {
...
-    template<typename PROD>
-    bool
-    getByLabel(InputTag const& tag, Handle<PROD>& result) const;
+    template <typename PROD>
+    bool getByLabel(InputTag const& tag, Handle<PROD>& result) const;
 
   private:
-    virtual BasicHandle getByLabelImpl(std::type_info const& iWrapperType, std::type_info const& iProductType, const InputTag& iTag) const = 0;
-
+    virtual BasicHandle getByLabelImpl(std::type_info const& iWrapperType,
```
Comment (Dr15Jones) [inline FWCore/Common/interface/LuminosityBlockBase.h]: I do like this version better than the original.

File: FWCore/Common/interface/Provenance.h [hunk 3]
```diff
@@ -8,5 +8,5 @@ namespace edm {
   class ParameterSet;
   std::string moduleName(Provenance const& provenance);
   ParameterSet const& parameterSet(Provenance const& provenance);
-}
+}  // namespace edm
```
Comment (Dr15Jones) [inline FWCore/Common/interface/Provenance.h]: I do like the comment it adds

File: FWCore/Common/interface/RunBase.h [hunk 4]
```diff
@@ -36,36 +36,33 @@ namespace edm {
...
-      convert_handle(std::move(bh), result);  // throws on conversion error
-      if (result.failedToGet()) {
-         return false;
-      }
-      return true;
-   }
-
-}
+  template <typename T>
+  bool RunBase::getByLabel(InputTag const& tag, Handle<T>& result) const {
```
Comment (Dr15Jones) [inline FWCore/Common/interface/RunBase.h]: The return value no longer on its own line is not my favorite, but I can live with it.

File: FWCore/Common/interface/TriggerResultsByName.h [hunk 5]
```diff
@@ -46,11 +46,8 @@ namespace edm {
   class HLTPathStatus;
 
   class TriggerResultsByName {
-
   public:
-
-    TriggerResultsByName(TriggerResults const* triggerResults,
-                         TriggerNames const* triggerNames);
+    TriggerResultsByName(TriggerResults const* triggerResults, TriggerNames const* triggerNames);
```
Comment (Dr15Jones) [inline FWCore/Common/interface/TriggerResultsByName.h]: This time it put all arguments on the same line. What is the threshold for same line versus different lines?

File: FWCore/Framework/interface/Callback.h [hunk 6]
```diff
@@ -27,91 +27,84 @@
...
+    };
+
+    template <typename T,          // producer's type
+              typename TReturn,    // return type of the producer's method
+              typename TRecord,    // the record passed in as an argument
+              typename TDecorator  // allows customization using pre/post calls
+              = CallbackSimpleDecorator<TRecord> >
+    class Callback : public CallbackBase {
+    public:
+      using method_type = TReturn (T ::*)(const TRecord&);
```
Comment (Dr15Jones) [inline FWCore/Framework/interface/Callback.h]: Is there an option for controlling which side we put the `const`? I'm actually a fan of on the right hand side (so called _east const_ ).

File: FWCore/Framework/interface/ComponentFactory.h [hunk 7]
```diff
@@ -34,100 +34,96 @@
...
-template<> edm::eventsetup::ComponentFactory<_type_> const* edm::eventsetup::ComponentFactory<_type_>::get() \
-{ return &s_dummyfactory; } \
-  } } \
-typedef int componentfactory_get_needs_semicolon
+    };
+
+  }  // namespace eventsetup
+}  // namespace edm
+#define COMPONENTFACTORY_GET(_type_)                                                                      \
+  EDM_REGISTER_PLUGINFACTORY(edmplugin::PluginFactory<edm::eventsetup::ComponentMakerBase<_type_>*()>,    \
```
Comment (Dr15Jones) [inline FWCore/Framework/interface/ComponentFactory.h]: This version is more readable than the original.

**Comment by davidlange6** [comment]
at least the code checks passed this time..

File: FWCore/Framework/interface/DependentRecordImplementation.h [hunk 8]
```diff
@@ -32,41 +33,43 @@
...
 
-      template<class DepRecordT>
+      template <class DepRecordT>
       std::optional<DepRecordT> tryToGetRecord() const {
-        //Make sure that DepRecordT is a type in ListT
-        typedef typename boost::mpl::end< ListT >::type EndItrT;
-        typedef typename boost::mpl::find< ListT, DepRecordT>::type FoundItrT;
-        static_assert(! std::is_same<FoundItrT, EndItrT>::value, "Trying to get a Record from another Record where the second Record is not dependent on the first Record.");
+        // Make sure that DepRecordT is a type in ListT
+        typedef typename boost::mpl::end<ListT>::type EndItrT;
```
Comment (Dr15Jones) [inline FWCore/Framework/interface/DependentRecordImplementation.h]: Is there an option for changing `typedef` to `using`? Or is that more a clang tidy thing?

File: FWCore/Framework/interface/EDLooperBase.h [hunk 9]
```diff
@@ -27,25 +27,23 @@
...
-   Once all events have been processed or kStop was returned from 'duringLoop' or ProcessingController was told to go to an
-   'invalid' event then the method
-      Status endOfLoop(EventSetup const&, unsigned int iCounter)
-   will be called.  iCounter will be the number of loops which have been run in the job starting at index 0. If kContinue
-   is returned from endOfLoop then a new loop will begin else if kStop is called then the job will end.
-
- Like other modules, an EDLooperBase is called for 'beginJob', 'endJob', 'beginRun', 'endRun', 'beginLuminosityBlock' and
- 'endLuminosityBlock'.
+   NOTE: if you have no need of controlling exactly what 'next' event should be processed then you should instead
+ inherit from the subclass EDLooper and us it simplified interface for 'duringLoop'. 3) end of loop Once all events have
```
Comment (Dr15Jones) [inline FWCore/Framework/interface/EDLooperBase.h]: It changed the formatting of the comment. It did not preserve the alignment of the paragraph.

**Comment by Dr15Jones** [comment]
I still have more to go, but I wanted to do an intermediate comment dump.

File: FWCore/Utilities/interface/typelookup.h [hunk 10]
```diff
@@ -47,54 +47,60 @@
...
+     */
+    class NameRegistrar {
+    public:
+      NameRegistrar(const char* iTypeName, const std::type_info& iInfo);
+    };
+
+  }  // namespace typelookup
+}  // namespace edm
+
+#define TYPELOOKUP_METHODS(Tname)                    \
```
Comment (Dr15Jones) [inline FWCore/Utilities/interface/typelookup.h]: This is more readable.

File: FWCore/MessageService/interface/ELstatistics.h [hunk 11]
```diff
@@ -34,98 +33,89 @@
...
+      bool printAtTermination;
 
-  
-  // summarization( const ELstring & sumLines, const ELstring & sumLines )
-  // from base class
+      CMS_THREAD_SAFE static std::set<std::string> groupedCategories;  // 8/16/07 mf
+      static ELstring formSummary(ELmap_stats& stats);                 // 8/16/07 mf
 
-protected:
-  int            tableLimit;
```
Comment (Dr15Jones) [inline FWCore/MessageService/interface/ELstatistics.h]: I was never a fan of the two column approach of declaring member variables anyway.

**Comment by Dr15Jones** [comment]
I think I've hit my limit on what I can look at. For the future, we should definitely not convert so many files on one pull request.

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
@slava77 @perrotta you may also want to take a look at how clang format changed the code.

File: FWCore/Catalog/interface/SiteLocalConfig.h [hunk 12]
```diff
@@ -2,18 +2,18 @@
...
 // PUBLIC DEFINES
 // PUBLIC CONSTANTS
 // PUBLIC TYPES
 namespace edm {
-    class ParameterSet;
-    class ActivityRegistry;
-}
+  class ParameterSet;
+  class ActivityRegistry;
+}  // namespace edm
```
Comment (wddgit) [inline FWCore/Catalog/interface/SiteLocalConfig.h]: For namespaces that include many lines, this added comment is nice. But for short ones like this, it just adds clutter.

File: FWCore/Common/interface/EventBase.h [hunk 13]
```diff
@@ -37,82 +37,79 @@
...
+    result.clear();
+    BasicHandle bh = this->getImpl(typeid(T), pid);
+    convert_handle(std::move(bh), result);  // throws on conversion error
+    if (result.failedToGet()) {
+      return false;
+    }
+    return true;
+  }
+
+}  // namespace edm
```
Comment (wddgit) [inline FWCore/Common/interface/EventBase.h]: It is hard to see what actually changed when indentation changes a large block of code and there are changes mixed in that only affect individual lines.

File: FWCore/Concurrency/interface/SerialTaskQueueChain.h [hunk 14]
```diff
@@ -68,98 +63,92 @@ namespace edm {
...
-    if(m_queues.size() == 1) {
-      m_queues[0]->push( [this,iAction]() mutable {this->actionToRun(iAction);} );
+    if (m_queues.size() == 1) {
+      m_queues[0]->push([this, iAction]() mutable { this->actionToRun(iAction); });
     } else {
       assert(!m_queues.empty());
-      m_queues[0]->push([this, iAction]() mutable {
-        this->passDownChain(1, iAction);
-      });
+      m_queues[0]->push([this, iAction]() mutable { this->passDownChain(1, iAction); });
```
Comment (wddgit) [inline FWCore/Concurrency/interface/SerialTaskQueueChain.h]: I find this harder to read. Putting the body of a lambda on a separate line seems better to me.

**Comment by davidlange6** [comment]
you can ignore whitespace changes in the GitHub diff viewer page by adding ?w=1 (or 0 I forget which..) to the url

File: FWCore/Framework/interface/DependentRecordImplementation.h (hunk 8 above)
Comment (davidlange6) [reply FWCore/Framework/interface/DependentRecordImplementation.h]: thats clang-tidy.

File: FWCore/Common/interface/EventBase.h (hunk 13 above)
Comment (davidlange6) [reply FWCore/Common/interface/EventBase.h]: as far as I see, we can either
1) add it everywhere
2) leave people to do it or not by their own judgement. 

I don't see an option to change according to length of namespace declaration

File: FWCore/Concurrency/interface/SerialTaskQueueChain.h (hunk 14 above)
Comment (davidlange6) [reply FWCore/Concurrency/interface/SerialTaskQueueChain.h]: you and @Dr15Jones make the same comment. We can forbid short functions from being on the same line - however I believe this change came out of a previous iteration (to be confirmed)

**Review by makortel — COMMENTED** [review]
Some notes from a non-complete look. Mostly I like the style.

File: FWCore/Concurrency/interface/SerialTaskQueueChain.h (hunk 14 above)
Comment (davidlange6) [reply FWCore/Concurrency/interface/SerialTaskQueueChain.h]: whops, I misread - so we already had this

AllowShortIfStatementsOnASingleLine: false
AllowShortLoopsOnASingleLine: false

but this is all one command, just a complex one. I'm looking into if there is a hook or not

**Comment by wddgit** [comment]
I spent the entire morning looking at this and I'm not even half way through ... I'm stopping at least for today. I did not notice anything other than more examples of what I or someone else already commented on. I did not notice anything that would break anything and create bugs. Overall I would say it is an improvement and makes the code easier to read. There is nothing that is so bad that I couldn't live with it or feel strongly about it. Kind of sad what it will do to attempts to trace history of code or determine attribution for who wrote what code ...  The information will all be there, but harder to find. Whoever runs this formatter is going to look like a coding genius to whoever runs "blame" in the GitHub code viewer, or maybe get the blame for all the bugs and all the questions. (Maybe we should create a git account for someone named "Clang Formatter").

If there is some piece of code where you do not like what this formatter is doing, is there a way to tell it to leave that piece of code alone? Also, will the rules enforced  by this formatter be documented in the style guide. Or maybe it just needs to include a link to the google documentation of the tool and a link to the file with our parameter settings. Hopefully this tool is stable and is not going to be changing the format of our code every year.

File: FWCore/Framework/interface/Callback.h (hunk 6 above)
Comment (davidlange6) [reply FWCore/Framework/interface/Callback.h]: belated answer - this is what clang-tidy might do for us. not clang-format

File: FWCore/Framework/interface/EDLooperBase.h (hunk 9 above)
Comment (davidlange6) [reply FWCore/Framework/interface/EDLooperBase.h]: I can try to turn off comment reformatting (seems this is possible)

File: FWCore/Catalog/interface/SiteLocalConfig.h (hunk 12 above)
Comment (davidlange6) [reply FWCore/Catalog/interface/SiteLocalConfig.h]: this option is all or none

**Comment by davidlange6** [comment]
next iteration is #25893

**Pull request closed without merging**